tree/treeplayer/src/TTreePlayer.cxx

```diff
@@ -733,18 +733,20 @@ Int_t TTreePlayer::MakeClass(const char *classname, const char *option)
    opt.ToLower();
 
    // Connect output files
-   if (!classname) classname = fTree->GetName();
-   const TString vclassname = ROOT::Internal::GetCppName(classname);
+   const TString fileNameStem = classname ? classname : fTree->GetName();
+   const TString cppClassName = ROOT::Internal::GetCppName(fileNameStem);
+   if (cppClassName != fileNameStem)
+      Warning("TTreePlayer::MakeClass", "The %s name provided ('%s') is not a valid C++ identifier and will be converted to '%s', the code produced will likely fail to compile.",(classname ? "class" : "tree"), fileNameStem.Data(), cppClassName.Data());
```

Comment:
Since we now convert both the case where the name comes from the `TTree` and from the argument the code should always compile, right?
```suggestion
      Warning("TTreePlayer::MakeClass", "The %s name provided ('%s') is not a valid C++ identifier and will be converted to '%s'.",(classname ? "class" : "tree"), fileNameStem.Data(), cppClassName.Data());
```